src/lib/db/public-signup-token-store.ts

```diff
@@ -30,22 +30,32 @@ interface ITokenUserRow {
     user_id: number;
     created_at: Date;
 }
+
 const tokenRowReducer = (acc, tokenRow) => {
-    const { userId, name, ...token } = tokenRow;
+    const { userId, userName, userUsername, roleId, roleName, ...token } =
+        tokenRow;
     if (!acc[tokenRow.secret]) {
         acc[tokenRow.secret] = {
             secret: token.secret,
             name: token.name,
+            url: token.url,
             expiresAt: token.expires_at,
             createdAt: token.created_at,
             createdBy: token.created_by,
-            roleId: token.role_id,
+            role: {
+                id: roleId,
+                name: roleName,
+            },
             users: [],
         };
     }
     const currentToken = acc[tokenRow.secret];
     if (userId) {
-        currentToken.users.push({ userId, name });
+        currentToken.users.push({
+            id: userId,
+            name: userName,
```

Comment:
What is the difference between userName and userUserName? Could we find a better naming?

Reply:
userName = the name of the user 
userUsername = the username of the user